pygotham/frontend/__init__.py

```diff
@@ -61,6 +64,51 @@ def create_app(settings_override=None):
     def current_event():
         return {'current_event': get_current_event()}
 
+    @app.context_processor
+    def generate_navbar():
+        """Autodiscover links that should populate the site's navbar."""
+        # navbar_links is a dict of the form
+        # {section_name: {link_name: link_value, ...}, ...}
+        navbar_links = defaultdict(dict)
+
+        # First, autogenerate a list based on available routes
+        # Available routes are any that have a GET method and have all
+        # arguments (if any) provided default values
+        for rule in app.url_map.iter_rules():
+            section_name = rule.endpoint.split('.')[0]
+            get_allowed = 'get' in (method.lower() for method in rule.methods)
+            required_args = set(rule.arguments)
+            provided_args = set(getattr(rule.defaults, 'keys', lambda: [])())
+            missing_args = required_args.difference(provided_args)
+            if get_allowed and not missing_args:
+                rule_name = rule.endpoint.split('.')[1]
+                navbar_links[section_name][rule_name] = url_for(rule.endpoint)
+
+        # Find module-specific overrides and update the routes
+        for _, name, _ in pkgutil.iter_modules(__path__):
+            m = importlib.import_module('{}.{}'.format(__name__, name))
+            if hasattr(m, 'get_nav_links'):
+                for section, override_links in m.get_nav_links().items():
+                    navbar_links[section].update(override_links)
+
+        # Normalize, hoist, and sort
+        # Exclude certain sections whose links are exposed elsewhere
+        excluded_sections = ('home', 'security', 'profile')
+        nav_list = []
+        for section, links in navbar_links.items():
+            if section in excluded_sections:
+                continue
+            subnav_list = []
+            index_link = links.pop('index', None) or links.pop('home', None)
+            for title, link in links.items():
+                subnav_list.append((title.replace('_', ' ').title(), link))
+            subnav_list.sort(key=lambda item: item[0])
+            if index_link:
+                subnav_list.insert(0, ('Home', index_link))
```

Comment:
Why not do this before the for loop? Then it's just an append rather than an insert.